packages/contract/src/contract-factory.ts

```diff
@@ -154,18 +200,135 @@ export default class ContractFactory {
 
+    const transactionResponse = await account.sendTransaction(transactionRequest);
+
+    const waitForResult = async () => {
+      const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
+      const contract = new Contract(contractId, this.interface, account) as TContract;
+
+      return { contract, transactionResult };
+    };
+
+    return { waitForResult, contractId, transactionId: transactionResponse.id };
+  }
+
+  /**
+   * Deploy a contract with the specified options.
+   *
+   * @param deployContractOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContractLoader<TContract extends Contract = Contract>(
+    deployContractOptions: DeployContractOptions = {}
+  ): Promise<DeployContractResult<TContract>> {
     const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
 
-    const transactionResponse = await account.sendTransaction(transactionRequest);
+    // Ensure the max contract size is byte aligned before chunking
+    if (maxContractSize % WORD_SIZE !== 0) {
+      throw new FuelError(
+        ErrorCode.CONTRACT_SIZE_EXCEEDS_LIMIT, // Todo: change error
+        `Max contract size of ${maxContractSize} bytes is not byte aligned.`
+      );
+    }
+
+    const { configurableConstants } = deployContractOptions;
+    if (configurableConstants) {
+      this.setConfigurableConstants(configurableConstants);
+    }
+
+    // Find the max chunk size (again, ensure is word aligned)
+    const baseAssetId = account.provider.getBaseAssetId();
+    const fakeFunding: CoinQuantity[] = [{ assetId: baseAssetId, amount: bn(100_000) }];
+    const baseBlobRequest = this.blobTransactionRequest({ blobBytecode: randomBytes(32) });
+    baseBlobRequest.fundWithFakeUtxos(fakeFunding, baseAssetId);
+    const maxChunkSize = maxContractSize - baseBlobRequest.byteLength() - WORD_SIZE;
+    const padding = maxChunkSize % WORD_SIZE;
+    const chunkSize = padding + ((WORD_SIZE - (padding % WORD_SIZE)) % WORD_SIZE) + maxChunkSize;
+
+    // Chunk the bytecode
+    const chunks: ContractChunk[] = [];
+    for (let offset = 0, index = 0; offset < this.bytecode.length; offset += chunkSize, index++) {
+      const chunk = this.bytecode.slice(offset, offset + chunkSize);
+
+      // Chunks must be byte aligned
+      if (chunk.length % WORD_SIZE !== 0) {
+        const paddingLength = chunk.length % WORD_SIZE;
+        const paddedChunk = concat([chunk, new Uint8Array(paddingLength)]);
+        chunks.push({ id: index, size: paddedChunk.length, bytecode: paddedChunk });
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      chunks.push({ id: index, size: chunk.length, bytecode: chunk });
```

Comment:
```suggestion
      let chunk = this.bytecode.slice(offset, offset + chunkSize);
      let length = chunk.length

      // Chunks must be byte aligned
      if (chunk.length % WORD_SIZE !== 0) {
        length = chunk.length % WORD_SIZE;
        chunk = concat([chunk, new Uint8Array(length)]);
      }
      chunks.push({ id: index, size: chunk.length, bytecode: chunk });
```

Reply:
Implemented in `packages/contract/src/loader/utils.ts`.